src/python/zquantum/qaoa/problems/generators.py

```diff
@@ -41,3 +41,36 @@ def get_random_hamiltonians_for_problem(
         hamiltonians.append(hamiltonian)
 
     return hamiltonians
+
+
+def get_random_hamiltonian(
```

Comment:
**Request:** I would add `number_of_terms` argument – this way we have some control over how big the Hamiltonian will actually be, without the need for a `while` loop.

Reply:
How important is it that we get to control the size? I went with a while loop because that made things easier lol, but I'm happy to change it. 

For example: if `number_of_terms = 4`, `n_qubits = 8`, and `max_number_of_qubits_per_term = 2`, then it has to return 4 disjoint terms. How to design the code structure to do stuff like that in a clean way? Another one: what is a clean way to make sure every qubit is added to the random hamiltonian? (The methods I've tried all add a disproportionately large amount of conditional logic 😅) Any suggestions?

Reply:
I think it's not important to make sure that every qubit is added to the random Hamiltonian. It's random after all, so if the randomness decides that given qubit should be empty, let it be.

In order to make sure that the number of qubits actually fits, we could have the first term be expanded to include all the qubits – i.e. if we want to have 4 qubits and it's a 2 qubit term involving `Z0 * Z2`, we would expand it to be `Z0 * I1 * Z2 * I3`. 
Then we have an object that "uses" all the qubits, but doesn't necessarily includes operations on all the qubits?
WDYT?

Reply:
Hmm good point on not needing to make sure every qubit is added. I can try it out

Reply:
Fixed!!